flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFileFormatFactory.java

```diff
@@ -109,17 +110,21 @@ public CsvBulkDecodingFormat(ReadableConfig formatOptions) {
             final RowType physicalRowType = (RowType) physicalDataType.getLogicalType();
             final CsvSchema schema = buildCsvSchema(physicalRowType, formatOptions);
 
+            final boolean ignoreParseErrors =
+                    formatOptions.getOptional(IGNORE_PARSE_ERRORS).isPresent();
             final Converter<JsonNode, RowData, Void> converter =
                     (Converter)
-                            new CsvToRowDataConverters(false)
+                            new CsvToRowDataConverters(ignoreParseErrors)
                                     .createRowConverter(projectedRowType, true);
-            return new StreamFormatAdapter<>(
+            CsvReaderFormat<RowData> csvReaderFormat =
                     new CsvReaderFormat<>(
                             new CsvMapper(),
                             schema,
                             JsonNode.class,
                             converter,
-                            context.createTypeInformation(projectedDataType)));
+                            context.createTypeInformation(projectedDataType));
+            csvReaderFormat.setIgnoreParseErrors(ignoreParseErrors);
```

Comment:
Nit: WDYT about passing the complete configuration to the `Reader` class? It feels more extensible in case we also need to pass other configurations to the `Reader`.

The setter, in this case, is only invoked one time and it is similar to a builder method.

Reply:
It is a bit unclear what do you propose? Do you mean adding something like a `ReaderConfig` with a builder and passing it into the `CsvReaderFormat` instead of the individual parameters? 
As mentioned in my previous comment, I decided to make it a setter because otherwise, we would need to expand all of the static factory methods (`fromShcma`, `fromPojo`) and "pollute" them with this special concern. In that case I would add utility "telescope" methods with this parameter set to "ignore-parse-errors=false" by default.

Reply:
Good point. I think the best way here is to use the `wither` pattern [1]. You can add a method like

`public CsvReaderFormat<T> withIgnoreParseErrors()`

that creates a new `CsvReaderFormat` with the flag that it ignores the errors. By solely using `fromSchema` or `fromPojo` nothing changes. Users can now do `CsvReaderFormat.fromPojo(Pojo.class).withIgnoreParseErrors()`

[1] https://projectlombok.org/features/With